pkg/cloudprovider/aws/fake/ec2api.go

```diff
@@ -36,82 +37,88 @@ type EC2API struct {
 	DescribeInstanceTypeOfferingsOutput *ec2.DescribeInstanceTypeOfferingsOutput
 	DescribeAvailabilityZonesOutput     *ec2.DescribeAvailabilityZonesOutput
 	WantErr                             error
+	CalledWithCreateFleetInput          []ec2.CreateFleetInput
+	Instances                           []*ec2.Instance
+}
 
-	CalledWithCreateFleetInput []ec2.CreateFleetInput
-	Instances                  []*ec2.Instance
+type EC2API struct {
+	ec2iface.EC2API
+	EC2Behavior
 }
 
-func (a *EC2API) Reset() {
-	a.CalledWithCreateFleetInput = nil
+// Reset must be called between tests otherwise tests will pollute
+// each other.
+func (e *EC2API) Reset() {
+	e.EC2Behavior = EC2Behavior{}
```

Comment:
Idk if this is crazy, but is it possible to keep this as 
```
func (a *EC2API) {
  a = &EC2API{}
}
```

Reply:
That is on the face of it some rather scary Go code, so I would be hesitant to go that route.

But also there's no guarantee that other code is not relying on the same pointer to the ec2api. In fact there is code like:

```go
	fakeEC2API = &fake.EC2API{}
	subnetProvider := &SubnetProvider{
		ec2api: fakeEC2API,
		cache:  subnetCache,
	}
```

Reply:
I guess inline behavior is pretty clean :)